tracing: update LightStep tracer

rnburn opened the pull request:
Description:

Updates to the a newer version of LightStep's tracer. Much of the tracer was rewritten to provide better performance.

In our [benchmarking](https://github.com/lightstep/lightstep-tracer-cpp/blob/8b7f871242a5d9f514a6eddf0ae51c15749d5ce7/benchmark/manual_tracer_benchmark.cpp), the speedup is around [2x-5x](https://7739-57146219-gh.circle-artifacts.com/0/benchmark/manual_tracer_benchmark_result.txt)

Risk Level: Medium
Testing:
Ran an [example](https://github.com/rnburn/envoy-lightstep-example) and verified that it reports traces.

dio commented:
Seems like you need to merge master?

moderation commented:
Does this fix https://github.com/envoyproxy/envoy/issues/7363 ?

moderation commented on bazel/repository_locations.bzl:
```diff
@@ -129,9 +129,9 @@ REPOSITORY_LOCATIONS = dict(
-        urls = ["https://github.com/lightstep/lightstep-tracer-cpp/archive/v0.8.0.tar.gz"],
+        sha256 = "9dc52831b2ea5b9434bdd5d82850a6b24142e74b042d9bda8fa0532cdc8b4d51",
+        strip_prefix = "lightstep-tracer-cpp-66b88418ca5190eeb6131e2e021fdb7c2f3239ad",
+        urls = ["https://github.com/lightstep/lightstep-tracer-cpp/archive/66b88418ca5190eeb6131e2e021fdb7c2f3239ad.tar.gz"],
```
As you are picking up a commit after the last 0.12.0 release please add a comment date for your commit `# 2020-02-24`.

rnburn replied:
Done

rnburn commented:
Note: LightStep's tracer doesn't do anything with the gRPC response message. Only the response codes are relevant for determining if a report request succeeded.

Since the response isn't used, I changed the code to ignore it and dropped the invalid response tests.

rnburn commented:
@lizan @moderation - could you guys take a look?

moderation commented:
From a patch removal and dependency management POV it LGTM. Will need to defer to @lizan for the code changes

moderation commented:
@rnburn I just built successfully for aarch64 but I get the following error when trying to run. Any tips on how to debug further and get more detailed output appreciated.
```
[2020-03-11 18:38:24.851][16423][info][tracing] [source/common/tracing/http_tracer_manager_impl.cc:25] instantiating tracing driver: envoy.tracers.lightstep
fish: '/home/username/Library/envoy/…' terminated by signal SIGILL (Illegal instruction)
```

rnburn commented:
Does it have anything more in the stack trace?

moderation commented:
I'm compiling with the following in my `.bazelrc` file. I'm guessing I'll need to change to get more info. I pinged you on Slack in case that might be faster:
```
build --define disable_object_dump_on_signal_trace=disabled
build --define signal_trace=disabled
```
